### PR TITLE
[Replace Commands Branch] Optimize draw_sphere

### PR DESCRIPTION
**Pull request by altrisi**
This PR is for the branch of gnembon#553. It:

- Optimizes `draw_sphere` so instead of relying on the extremely expensive `break()` and `continue()`, it is just whether to continue in an if
   - For continue, the code is just not executed (instead of `if invalid continue()`, it does `if valid do`)
   - For break, a variable (`instaquit`) is initialized outside the loop. If the break condition is met, `instaquit = true`. If it is true, code is just skipped.
    - For return, since functions return by default whatever there is at the end of them, it just places it at the end without the return call (because it is also quite expensive)
- Moves common usages to a function
   - Basically every function calls a print to tell affected blocks, so let's make one for that (`affected(player)`)
   - Tracking affected blocks is now done via a global instead, that is updated from `set_block`, so it doesn't need to be calculated and reseted everywhere
   - Still "returns" the count of affected blocks by creating `affected` there: nano-optimization to say the most

Also removes the check to carpet rule (as discussed in a comment) since it's not needed and just double-updates and adds some overhead (since `carpet_rules` is also quite expensive and is ran for every block).

### CODE REVIEW

**Comment by altrisi** [comment]
There is still another `return()` statement in the code (at `_draw_line`) which could also be removed, but it's not used that much (still used in a function that is accumulated every run, so it may be good to remove it, up to you).

**Comment by Ghoulboy78** [comment]
OK great man thx
